tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/BmcOper.scala

```diff
@@ -64,7 +65,9 @@ object BmcOper {
 
   /**
     * The distinct operator that is equivalent to (distinct ...) in SMT-LIB.
-    * Formally, BMC!Distinct(x_1, ..., x_n) is equivalent to \A i, j \in 1..n: i /= j => x_i /= x_j
+    * Formally, BMC!Distinct(x_1, ..., x_n) is equivalent to \A i, j \in 1..n: i /= j => x_i /= x_j.
+    *
+    * XXX: there seems to be no way of defining a user-defined variadic operator in Apalache.tla.
```

Comment:
is this a feature we might upstream? Since you specify `user-defined` it makes it sound like this might be feasible in the model checker and AST, but just isn't supported at the level of parsing?

Reply:
It does not seem to be supported by the TLA+ parser. Apparently, variadic operators are like a special form in TLA+. They are like other operators, but not exactly.

Reply:
Thanks!